app/frontend/src/features/messages/surfing/HostRequestView.tsx

```diff
@@ -19,55 +21,53 @@ export interface HostRequestViewProps extends BoxProps {
 }
 
 export default function HostRequestView({ hostRequest }: HostRequestViewProps) {
-  const [loading, setLoading] = useState(false);
-  const [error, setError] = useState("");
-  const [messages, setMessages] = useState<Message.AsObject[]>([]);
-  const fetchMessages = useCallback(async () => {
-    setLoading(true);
-    try {
-      const messages = await service.requests.getHostRequestMessages(
-        hostRequest.hostRequestId
-      );
-      setMessages(messages);
-    } catch (error) {
-      setError(error.message);
-    }
-    setLoading(false);
-  }, [hostRequest.hostRequestId]);
-
-  useEffect(() => {
-    fetchMessages();
-  }, [fetchMessages]);
-
-  const handleSend = async (text: string) => {
-    await service.requests.sendHostRequestMessage(
-      hostRequest.hostRequestId,
-      text
-    );
-    await fetchMessages();
-  };
+  const { data: messages, isLoading, error } = useQuery<
+    Message.AsObject[],
+    GrpcError
+  >(["hostRequestMessages", hostRequest.hostRequestId], () =>
+    service.requests.getHostRequestMessages(hostRequest.hostRequestId)
+  );
 
   const { data: surfer, isLoading: surferLoading } = useUser(
     hostRequest.fromUserId
   );
   const { data: host, isLoading: hostLoading } = useUser(hostRequest.toUserId);
-  const { data: currentUser } = useCurrentUser();
-  const surferName =
-    currentUser?.userId === surfer?.userId ? "you" : surfer?.name;
-  const hostName = currentUser?.userId === host?.userId ? "you" : host?.name;
+  const currentUserId = useAuthContext().authState.userId;
+  const surferName = currentUserId === surfer?.userId ? "you" : surfer?.name;
+  const hostName = currentUserId === host?.userId ? "you" : host?.name;
   const title = `${surferName} requested to be hosted by ${hostName}`;
 
+  const queryClient = useQueryClient();
+  const mutation = useMutation<string | undefined, GrpcError, string>(
+    (text: string) =>
+      service.requests.sendHostRequestMessage(hostRequest.hostRequestId, text),
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries([
+          "hostRequestMessages",
+          hostRequest.hostRequestId,
+        ]);
+        queryClient.invalidateQueries(["hostRequests"]);
+      },
+    }
+  );
+
   const classes = useStyles();
   return (
     <Box className={classes.root}>
       <Typography variant="h3">
         {surferLoading || hostLoading ? <Skeleton /> : title}
       </Typography>
-      {error && <Alert severity={"error"}>{error}</Alert>}
-      {loading ? (
+      {(error || mutation.error) && (
+        <Alert severity={"error"}>{error?.message || mutation.error}</Alert>
+      )}
+      {isLoading ? (
         <CircularProgress />
       ) : (
-        <MessageList messages={messages} handleSend={handleSend} />
+        <>
+          <MessageList messages={messages!} />
```

Comment:
Think this would have the same problem I mentioned for `GroupChatsTab` if `messages` is `undefined` because of data loading error. I wonder if it's better to split `error` and `mutationError` into its own rendering bit so we can do something like this again:

```jsx
{isLoading ? (
  <CircularProgress />
) : error ? (
  <Alert severity={"error"}>{error?.message || mutation.error}</Alert>
) : (
  <>
    <MessageList ... />
  </>
)
```